### PR TITLE
feat(tron): chain package

### PR DESCRIPTION
**Pull request by vlnevyhosteny**
Added:

- @types/elliptic
- tronweb

### CODE REVIEW

File: packages/blockchain/tron/src/lib/services/tron.wallet.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    generateBlockchainWallet,
+    /**
+     * Generate wallet
+     * @param mnemonic optional mnemonic seed to use. If not present, new one will be generated
+     * @returns wallet or a combination of address and private key
+     */
+    async generateWallet(mnemonic?: string): Promise<TronWallet> {
+      mnemonic ||= generateMnemonic(256)
+
+      return generateBlockchainWallet(mnemonic)
```
Comment (npwork) [inline packages/blockchain/tron/src/lib/services/tron.wallet.ts:39]: maybe `generateBlockchainWallet(mnemonic ?? generateMnemonic(256))` ?

File: packages/blockchain/tron/src/lib/services/tron.web.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      const HttpProvider = TronWeb.providers.HttpProvider
+
+      const fullNode = new HttpProvider(endpoint)
+      const solidityNode = new HttpProvider(endpoint)
+      const eventServer = new HttpProvider(endpoint)
+
+      const tronWeb = new TronWeb(fullNode, solidityNode, eventServer)
+
+      tronWeb.setHeader({ 'TRON-PRO-API-KEY': process.env['TRON_PRO_API_KEY'] })
```
Comment (npwork) [inline packages/blockchain/tron/src/lib/services/tron.web.ts:26]: extract to const

File: packages/blockchain/tron/src/lib/tron.sdk.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const web = tronWeb()
+  const api = BlockchainTronService
+
+  return {
+    ...abstractBlockchainSdk({
+      ...args,
+      blockchain,
+    }),
+    transaction: tronTx({ tronWeb: web }),
+    api,
```
Comment (npwork) [inline packages/blockchain/tron/src/lib/tron.sdk.ts:22]: please remove api

File: packages/blockchain/tron/src/lib/tron.sdk.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    wallet: tronWallet({ tronWeb: web }),
+    tronWeb: web,
+    record: tronRecord(),
+    blockchain: {
+      broadcast: BlockchainTronService.tronBroadcast,
+      getCurrentBlock: BlockchainTronService.tronGetCurrentBlock,
+      getBlock: BlockchainTronService.tronGetBlock,
+      getTrc10Detail: BlockchainTronService.tronTrc10Detail,
+      getAccount: BlockchainTronService.tronGetAccount,
+      getTransaction: BlockchainTronService.tronGetTransaction,
```
Comment (npwork) [inline packages/blockchain/tron/src/lib/tron.sdk.ts:32]: Please check `btc.blockchain.spec.ts` - so you can add tests for mapping

**Comment by npwork** [comment]
Also there are [50 Code Smells](https://sonarcloud.io/project/issues?id=tatumio_tatum-js&pullRequest=357&resolved=false&types=CODE_SMELL) please fix them

File: packages/blockchain/tron/src/lib/services/tron.wallet.ts (hunk 1 above)
Comment (vlnevyhosteny) [reply packages/blockchain/tron/src/lib/services/tron.wallet.ts:39]: Thats better, fixed.

File: packages/blockchain/tron/src/lib/tron.sdk.ts (hunk 3 above)
Comment (vlnevyhosteny) [reply packages/blockchain/tron/src/lib/tron.sdk.ts:22]: Ok, but there is api for evm-based chains. Why not for tron?

Comment (npwork) [reply packages/blockchain/tron/src/lib/tron.sdk.ts:22]: Actually I want to get rid of API field from everywhere. we already have those methods in blockchain

File: packages/blockchain/tron/src/lib/services/tron.web.ts (hunk 2 above)
Comment (vlnevyhosteny) [reply packages/blockchain/tron/src/lib/services/tron.web.ts]: done

**Comment by vlnevyhosteny** [comment]
> Also there are [50 Code Smells](https://sonarcloud.io/project/issues?id=tatumio_tatum-js&pullRequest=357&resolved=false&types=CODE_SMELL) please fix them

Fixed them. Rest of them are TODOs caused by OpenAPI issues.

File: packages/blockchain/tron/src/lib/tron.sdk.ts (hunk 3 above)
Comment (vlnevyhosteny) [reply packages/blockchain/tron/src/lib/tron.sdk.ts:22]: Ok, got it.

File: packages/blockchain/tron/src/lib/tron.sdk.ts (hunk 4 above)
Comment (vlnevyhosteny) [reply packages/blockchain/tron/src/lib/tron.sdk.ts:30]: Done

**Review by npwork — APPROVED** [review]

**Pull request merged**